packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -3646,6 +3646,51 @@ describeWithDOM('mount', () => {
         expect(wrapper.prop('bar')).to.equal('bye');
       });
     });
+
+    describe('props in async handler', () => {
+      class TestComponent extends React.Component {
+        constructor(props) {
+          super(props);
+          this.state = { counter: 1 };
+          this.handleClick = this.handleClick.bind(this);
+          this.delay = this.delay.bind(this);
+        }
+
+        async handleClick() {
+          this.delay(100);
```

Comment:
without `await this.delay(100)`, this line has no effect. Try this instead:
```suggestion
          return this.delay(100).then(() => {
            return new Promise(…)
          });
```